rust_src/src/lisp.rs

```diff
@@ -745,6 +763,30 @@ impl Iterator for TailsIter {
     }
 }
 
+pub struct CarIter {
+    tails: TailsIter,
+}
+
+impl CarIter {
+    pub fn new(list: LispObject, errsym: Option<Lisp_Object>) -> Self {
+        Self {
+            tails: TailsIter::new(list, errsym),
+        }
+    }
+
+    pub fn current(&self) -> LispObject {
+        self.tails.tail
+    }
+}
+
+impl Iterator for CarIter {
+    type Item = LispObject;
+
+    fn next(&mut self) -> Option<Self::Item> {
+        self.tails.next().and_then(|c| Some(c.car())).or(None)
```

Comment:
That looks like it should just be `.map(|c| c.car())`?

Reply:
Yes. I was not sure which read better.